File: backend/shared/uidl/src/main/java/io/mateu/uidl/data/ConfirmDialog.java
package io.mateu.uidl.data;

import io.mateu.uidl.fluent.Component;
import lombok.Builder;

@Builder
public record ConfirmDialog(
    String header,
    Component content,
    boolean canCancel,
    boolean canReject,
    String rejectText,
    String confirmText,
    String openedCondition,
    String confirmActionId,
    String rejectActionId,
    String cancelActionId,
    String style,
    String cssClasses)
    implements Component {}

File: frontend/web/monorepo/libs/mateu/src/mateu/shared/apiClients/dtos/componentmetadata/ConfirmDialog.ts
import ComponentMetadata from "@mateu/shared/apiClients/dtos/ComponentMetadata";
import Component from "@mateu/shared/apiClients/dtos/Component";

export default interface ConfirmDialog extends ComponentMetadata {

    header: string
    content: Component
    canCancel: boolean
    canReject: boolean
    rejectText: string
    confirmText: string
    openedCondition: string
    confirmActionId: string
    rejectActionId: string
    cancelActionId: string

}